src/Apps/Consign/Routes/SubmissionFlow/ArtworkDetails/Components/ArtworkDetailsForm.tsx

```diff
@@ -24,34 +24,61 @@ import {
 import { useFormikContext } from "formik"
 import { compact } from "lodash"
 import { useState } from "react"
+import { ArtworkDetails_myCollectionArtwork } from "__generated__/ArtworkDetails_myCollectionArtwork.graphql"
 import { ArtworkDetails_submission } from "__generated__/ArtworkDetails_submission.graphql"
 import { postalCodeValidators } from "../../Utils/validation"
 import { ArtistAutoComplete } from "./ArtistAutocomplete"
 
 export const getArtworkDetailsFormInitialValues = (
-  submission?: ArtworkDetails_submission
-): ArtworkDetailsFormModel => ({
-  artistId: submission?.artist?.internalID ?? "",
-  artistName: submission?.artist?.name ?? "",
-  year: submission?.year ?? "",
-  title: submission?.title ?? "",
-  materials: submission?.medium ?? "",
-  rarity: submission?.attributionClass?.replace("_", " ").toLowerCase() ?? "",
-  editionNumber: submission?.editionNumber ?? "",
-  editionSize: submission?.editionSize ?? undefined,
-  height: submission?.height ?? "",
-  width: submission?.width ?? "",
-  depth: submission?.depth ?? "",
-  units: submission?.dimensionsMetric ?? "in",
-  provenance: submission?.provenance ?? "",
-  location: {
-    city: submission?.locationCity ?? "",
-    country: submission?.locationCountry ?? undefined,
-    state: submission?.locationState ?? undefined,
-    countryCode: submission?.locationCountryCode ?? undefined,
-  },
-  postalCode: submission?.locationPostalCode ?? undefined,
-})
+  submission?: ArtworkDetails_submission,
```

Comment:
**suggestion:** I feel like this method can be improved a bit. I suggest few changes here: 
1. Improve the args to be specified explicitly from wherever this is being called

For example:
```typescript
getArtworkDetailsFormInitialValues({values, type})
```
2. Improve types to avoid optional chaining everywhere. We can do something like the following
3. 
```
export const getArtworkDetailsFormInitialValues = ({
  values,
  type,
}:
  | {
      values: ArtworkDetails_submission
      type: "SUBMISSION"
    }
  | {
      values: number[]
      type: "MY_COLLECTION_ARTWORK"
    }) => {
    // .... 
```

3. I would suggest as well avoiding long ternaries in this case but not replacing it with an if statement and instead use a switch statement with `assertNever` . Maybe something like this:

```
switch (type) {
   case "SUBMISSION":
       return {}
case "MY_COLLECTION_ARTWORK":
       return {}
default:
       assertNever(type)

}
```

Reply:
Hm, can't make it with the types
Let's revisit on some of the knowledge shared to not block the progress?

Reply:
sure, let's check it tomorrow morning since we will be talking about OKRs during the KS